File: web/app/Model/Asset.php
<?php

namespace App\Model;

use Illuminate\Database\Eloquent\Model;

use Illuminate\Http\UploadedFile;

class Asset extends Model
{
    protected $guarded = ['id'];
    protected $appends = ['link'];

    public static function createNewFile(UploadedFile $file, Array $attribute = [], $path = null, $disk = 'public') {
    	return static::create(array_merge([
    		'name' => $file->disk($disk)->storeAs(
    			$path ?? 'assets',
    			\Str::random(40) . '.' . $file->getClientOriginalExtension()
    		)
    	], $attribute));
    }

    public function getLinkAttribute()
    {
    	return asset('assets/images/' . $this->name);
    }
}

File: mobile/lib/model/asset.dart
class Asset {
  int id;
  String name;
  String link;

  Asset({this.id, this.name, this.link});

  factory Asset.parseFromJson(Map<String, dynamic> map) => Asset(
    id: map['id'],
    name: map['name'],
    link: map['link']
  );
}